### PR TITLE
Create afnor690-2010-uco.csl

### PR DESCRIPTION
**Pull request by Kalunna**
Hello,
Here is a French adaptation of the AFNOR 690 version of the year 2010, for the Catholic University of the West.
Thanks

### CODE REVIEW

File: afnor690-2010-uco.csl [hunk 1]
```diff
@@ -0,0 +1,628 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" version="1.0" class="in-text" demote-non-dropping-particle="never">
+  <!-- Standards AFNOR 690 Z 44-005 French adaptation 2010. Catholic University of the West Angers- Libraries -->
+  <info>
+    <title>AFNOR-690 édition 2010 (Université catholique de l'Ouest)</title>
+    <title-short>AFNOR-690 for UCO</title-short>
+    <id>http://www.zotero.org/styles/afnor690-2010-uco</id>
+    <link href="http://www.zotero.org/styles/afnor690-2010-uco" rel="self"/>
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:8]: Please add a "template" link for the style you started with to make this adaptation.

File: afnor690-2010-uco.csl [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+        <name-part name="given"/>
+      </name>
+    </names>
+  </macro>
+  <macro name="responsability">
+    <choose>
+      <if type="patent">
+        <!-- FIXME
+Patent main creator should be Zotero "assignee", which is not yet mapped against any CSL field
+-->
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:64]:
```suggestion
        <!-- TODO: Patent main creator should be Zotero "assignee", which is not yet available in CSL -->
```

File: afnor690-2010-uco.csl [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+          </if>
+          <else-if variable="editor">
+            <text macro="editor"/>
+          </else-if>
+        </choose>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="patent-inventor">
+    <!-- Called by secondary-responsability-->
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:79]:
```suggestion
    <!-- Called by secondary-responsibility-->
```

File: afnor690-2010-uco.csl [hunk 4]
```diff
@@ -0,0 +1,628 @@
...
+    <text value="Inventeur(s)" suffix="&#8239;: "/>
+    <names variable="author">
+      <name and="text" sort-separator=" " delimiter=", " delimiter-precedes-last="never">
+        <name-part name="family" text-case="uppercase"/>
+        <name-part name="given"/>
+      </name>
+    </names>
+  </macro>
+  <macro name="secondary-responsability">
+    <!-- FIXME : improve management of secondary responsabiliy for other types -->
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:89]:
```suggestion
  <macro name="secondary-responsibility">
    <!-- FIXME : improve management of secondary responsibiliy for other types -->
```

File: afnor690-2010-uco.csl [hunk 5]
```diff
@@ -0,0 +1,628 @@
...
+  </macro>
+  <macro name="container-author">
+    <names variable="container-author">
+      <name and="text" name-as-sort-order="all" sort-separator=", " delimiter=", " delimiter-precedes-last="never">
+        <name-part name="family" text-case="uppercase"/>
+        <name-part name="given"/>
+      </name>
+    </names>
+  </macro>
+  <macro name="container-responsability">
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:105]:
```suggestion
  <macro name="container-responsibility">
```

File: afnor690-2010-uco.csl [hunk 6]
```diff
@@ -0,0 +1,628 @@
...
+    <names variable="editor">
+      <name and="text" name-as-sort-order="all" sort-separator=", " delimiter=", " delimiter-precedes-last="never">
+        <name-part name="family" text-case="uppercase"/>
+        <name-part name="given"/>
+      </name>
+      <label prefix=" (" form="short" suffix=".)"/>
+    </names>
+  </macro>
+  <macro name="translator">
+    <!-- Called by secondary-responsability-->
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:50]:
```suggestion
    <!-- Called by secondary-responsibility-->
```

File: afnor690-2010-uco.csl [hunk 7]
```diff
@@ -0,0 +1,628 @@
...
+    <names variable="translator">
+      <label text-case="capitalize-first" form="verb-short" suffix=" "/>
+      <name and="text" sort-separator=" " delimiter=", " delimiter-precedes-last="never">
+        <name-part name="family" text-case="uppercase"/>
+        <name-part name="given"/>
+      </name>
+    </names>
+  </macro>
+  <macro name="responsability">
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:59]:
```suggestion
  <macro name="responsibility">
```

File: afnor690-2010-uco.csl [hunk 8]
```diff
@@ -0,0 +1,628 @@
...
+          <if variable="container-title">
+            <text variable="title" suffix=". "/>
+            <choose>
+              <if type="article-journal article-magazine article-newspaper" match="none">
+                <text term="in" text-case="capitalize-first" suffix="&#8239;: "/>
+              </if>
+            </choose>
+            <choose>
+              <if variable="container-author editor" match="any">
+                <text macro="container-responsability"/>
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:132]:
```suggestion
                <text macro="container-responsibility"/>
```

File: afnor690-2010-uco.csl [hunk 9]
```diff
@@ -0,0 +1,628 @@
...
+      </choose>
+      <text macro="archive"/>
+    </group>
+  </macro>
+  <macro name="note">
+    <text variable="note" text-case="capitalize-first"/>
+  </macro>
+  <macro name="citation-reference">
+    <group delimiter=". ">
+      <text macro="responsability"/>
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:546]:
```suggestion
      <text macro="responsibility"/>
```

File: afnor690-2010-uco.csl [hunk 10]
```diff
@@ -0,0 +1,628 @@
...
+          <!-- volume : used for books and monographs (a volume of a multivolume book)-->
+          <text macro="volume"/>
+        </group>
+        <!-- medium : [online], [map], [software], etc.-->
+        <text macro="medium_designation"/>
+      </group>
+      <text macro="thesis-report-genre"/>
+      <text macro="edition"/>
+      <!-- secondary-responsability : used for patent inventors and translators -->
+      <!--  <text macro="secondary-responsability"/> -->
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:559]:
```suggestion
```

File: afnor690-2010-uco.csl [hunk 11]
```diff
@@ -0,0 +1,628 @@
...
+        </if>
+        <else>
+          <text macro="url"/>
+        </else>
+      </choose>
+      <text macro="archive"/>
+    </group>
+  </macro>
+  <macro name="note">
+    <text variable="note" text-case="capitalize-first"/>
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:542]: What is the rationale here? `"note"` should generally only be included for annotated bibliographies as it often contains a lot of information that should not be cited.

File: afnor690-2010-uco.csl [hunk 12]
```diff
@@ -0,0 +1,628 @@
...
+          <text macro="numberOfVolumes"/>
+          <text macro="issue-pagination-locator"/>
+        </group>
+      </group>
+      <text macro="accessed"/>
+      <text macro="access"/>
+      <text macro="note"/>
+    </group>
+  </macro>
+  <macro name="bibliography-reference">
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:579]: Please remove the `citation-reference` and `bibliography-reference` macros and just put the contents directly in the `layout` section of `citation` and `bibliography`.

File: afnor690-2010-uco.csl [hunk 13]
```diff
@@ -0,0 +1,628 @@
...
+    <group delimiter=". ">
+      <text macro="responsability"/>
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:581]:
```suggestion
      <text macro="responsibility"/>
```

File: afnor690-2010-uco.csl [hunk 14]
```diff
@@ -0,0 +1,628 @@
...
+          <text macro="title"/>
+          <!-- volume : used for books and monographs (a volume of a multivolume book)-->
+          <text macro="volume"/>
+        </group>
+        <!-- medium : [online], [map], [software], etc.-->
+        <text macro="medium_designation"/>
+      </group>
+      <text macro="edition"/>
+      <!-- secondary-responsability : used for patent inventors and translators -->
+      <text macro="secondary-responsability"/>
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:593]:
```suggestion
      <!-- secondary-responsibility : used for patent inventors and translators -->
      <text macro="secondary-responsibility"/>
```

File: afnor690-2010-uco.csl [hunk 15]
```diff
@@ -0,0 +1,628 @@
...
+    </group>
+  </macro>
+  <citation et-al-min="4" et-al-use-first="3">
+    <layout suffix="." delimiter="&#8239;; ">
+      <text macro="citation-reference"/>
+    </layout>
+  </citation>
+  <bibliography>
+    <sort>
+      <key macro="responsability"/>
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:621]:
```suggestion
      <key macro="responsibility"/>
```

File: afnor690-2010-uco.csl [hunk 16]
```diff
@@ -0,0 +1,628 @@
...
+        <group delimiter=". ">
+          <group delimiter="&#8239;: ">
+            <text macro="publisher-place"/>
+            <text macro="publisher"/>
+          </group>
+          <text macro="number"/>
+        </group>
+        <group delimiter=", ">
+          <text macro="date"/>
+          <text macro="numberOfVolumes"/>
```
Comment (bwiernik) [inline afnor690-2010-uco.csl:570]: For consistency of style, please rename the macro `number-of-volumes`.

**Review by bwiernik — CHANGES_REQUESTED** [review]
Thanks! A few comments.

**Comment by adam3smith** [comment]
@Kalunna -- please see the above comments

**Comment by adam3smith** [comment]
Closing due to inactivity -- we're happy to re-open this if you find time to come back to it: simply post a short message here.

**Pull request closed without merging**